src/pyodide/internal/workers.py

```diff
@@ -604,3 +605,24 @@ async def json(self, **kwargs: Any) -> Any:
     async def text(self) -> str:
         self._raise_if_failed()
         return await self.js_object.text()
+
+
+def event(func):
+    """
+    When applied to handlers such as `on_fetch` it will rewrite arguments passed in to native Python
+    types defined in this module. For example, the `request` argument to `on_fetch` gets converted
+    to an instance of the Request class defined in this module.
+    """
+
+    @functools.wraps(func)
```

Comment:
I guess `functools.wraps` preserves the signature for introspection, so we can still tell whether the function wanted 3, 2, or 1 argument. In your example, the handler only takes one argument so we can see that this is working.